modules/configuring-the-codeflare-operator.adoc

```diff
@@ -0,0 +1,83 @@
+:_module-type: PROCEDURE
+
+[id="configuring-the-codeflare-operator_{context}"]
+= Configuring the CodeFlare Operator
+
+[role='_abstract']
+If you want to change the default configuration of the CodeFlare Operator for distributed workloads in {productname-short}, you can edit the associated config map.
+
+.Prerequisites
+ifdef::upstream,self-managed[]
+* You have logged in to {openshift-platform} with the `cluster-admin` role.
+endif::[]
+ifdef::cloud-service[]
+* You have logged in to OpenShift with the `cluster-admin` role.
+endif::[]
+
+
+.Procedure
+ifdef::upstream,self-managed[]
+. In the {openshift-platform} console, click *Workloads* -> *ConfigMaps*.
+endif::[]
+ifdef::cloud-service[]
+. In the OpenShift console, click *Workloads* -> *ConfigMaps*.
+endif::[]
+
+ifdef::self-managed,cloud-service[]
+. From the *Project* list, select *redhat-ods-applications*.
+endif::[]
+ifdef::upstream[]
+. From the *Project* list, select *odh*.
+endif::[]
+
+. Search for the *codeflare-operator-config* config map, and click the config map name to open the *ConfigMap details* page.
+
+. Click the *YAML* tab to show the config map specifications.
+. In the `*data*` > `*config.yaml*` > `*kuberay*` section, you can edit the following entries:
++
+`*ingressDomain*`::
+This configuration option is null (`ingressDomain: ""`) by default.
+This option should remain null unless the Ingress Controller is not running on OpenShift.
+{productname-short} uses this value to generate the dashboard and client routes for every Ray Cluster, as shown in the following examples:
++
+.Example dashboard and client routes
+[source,bash,subs="+quotes"]
+----
+ray-dashboard-_<clustername>_-_<namespace>_._<your.ingress.domain>_
+ray-client-_<clustername>_-_<namespace>_._<your.ingress.domain>_
+----
++
+`*mTLSEnabled*`::
+This configuration option is enabled (`mTLSEnabled: true`) by default.
+When this option is enabled, the Ray Cluster pods create certificates that are used for mutual Transport Layer Security (mTLS), a form of mutual authentication, between Ray Cluster nodes.
```

Comment:
It also requires mTLS between the client and head node

Reply:
In order to connect to the Ray head node interactively, clients will need to download the generated certificate from the secret named `ca-secret-{cluster_name}` and generate the necessary certs for communicating and then set the requisite Ray environment variables. The easiest way to do this is using the following functions of the CodeFlare SDK after you cluster has been created and is up and running:
```python
import codeflare.sdk.utils.generate_cert as cf_cert_gen

cf_cert_gen.generate_tls_cert(cluster_name, namespace)
cf_cert_gen.export_env(cluster_name, namespace)
```